pilot/pkg/serviceregistry/mcp/controller.go

```diff
@@ -125,116 +126,203 @@ func (c *controller) List(typ resource.GroupVersionKind, namespace string) (out
 	return out, nil
 }
 
-// Apply receives changes from MCP server and creates the
-// corresponding config
-func (c *controller) Apply(change *sink.Change) error {
-	s, ok := c.schemas.Find(change.Collection)
-	if !ok || change.Collection == collections.IstioNetworkingV1Alpha3SyntheticServiceentries.Name().String() {
-		return fmt.Errorf("apply type not supported %s", change.Collection)
+func (c *controller) Handle(e event.Event) {
+	q := c.syncingMap[e.Source.Name()]
+	if q == nil {
+		log.Warnf("received event for unsupported collection: %s", e.Source.Name())
+		return
+	}
+
+	switch e.Kind {
+	case event.Reset:
```

Comment:
if we get a Reset then FullSync isn't there a period of time that we have no config which can be really bad? I am probably misunderstanding this

Reply:
Only for the current update. Basically what we're doing is caching the events between reset and fullsync. So until we get the fullsync, users of the `ConfigStore` API will see the old data before the reset.